PackageViewModel/PackageChooser/ShowLatestVersionQueryContext.cs

```diff
@@ -122,12 +122,15 @@ private async Task<List<T>> SearchOnServer(CancellationToken token)
                 {
                     _rawPackageSearchResouce = await _sourceRepository.GetResourceAsync<RawSearchResourceV3>(token);
                 }
-                
+
                 if (_rawPackageSearchResouce != null)
                 {
-                    var json = await _rawPackageSearchResouce.Search(searchText, _searchContext.Filter, CurrentPage * _pageSize, _pageSize, NullLogger.Instance, token);
-
-                    result = json.Select(s => s.FromJToken<PackageSearchMetadata>());
+                    // Don't run the cpu-bound operations on GUI thread
+                    result = await Task.Run(async () =>
+                    {
+                        var json = await _rawPackageSearchResouce.Search(searchText, _searchContext.Filter, CurrentPage * _pageSize, _pageSize, NullLogger.Instance, token);
+                        return json.Select(s => s.FromJToken<PackageSearchMetadata>()).ToList();
+                    }, token);
```

Comment:
FYI, only difference: the Task.Run and the ToList  on `json.Select(s => s.FromJToken<PackageSearchMetadata>()`